src/libraries/Common/src/System/Security/Cryptography/ECDiffieHellmanCng.ImportExport.cs

```diff
@@ -36,7 +37,14 @@ public override void ImportParameters(ECParameters parameters)
                     }
 
                     byte[] ecNamedCurveBlob = ECCng.GetNamedCurveBlob(ref parameters, ecdh: true);
-                    ImportKeyBlob(ecNamedCurveBlob, curve.Oid.FriendlyName, includePrivateParamerters);
+                    ImportKeyBlob(ecNamedCurveBlob, curve.Oid.FriendlyName, includePrivateParameters);
+                }
+                else if (curve.IsNamed && !hasPublicParameters && includePrivateParameters)
+                {
+                    // Import through PKCS8 so that the public key can be reconstructed from the
+                    // private key if the public ECPoint is not specified.
+
+                    ImportLimitedPrivateKeyBlob(parameters);
```

Comment:
FWIW, I suspect that building the blobs with Q = (0, 0) will make CNG just calculate it.  I'd be slightly worried about that for custom key providers (HSMs, et al), but since the Import methods on our types always throw away the key handle and make an ephemeral software key, I think it'd be stable enough.

Reply:
> but since the Import methods on our types always throw away the key handle and make an ephemeral software key, I think it'd be stable enough.

Does that also hold for S.S.C.Cng types as well that expose KSPs and the like?

Reply:
The CSP types have some weird "sometimes this is a new key, sometimes this is trying to update the existing key"; but I believe all of the other types (and thus all of the CNG types) are "Import detaches from the current key and makes an ephemeral one".  To create a persisted key version of an existing key one has to use CngKey.Import/CngKey.Create or import a PFX.

Reply:
Okay that does indeed work.